File: libs/go/events/activity.go
// Package events defines shared cross-service event payloads.
package events

import "time"

// ActivityCreated represents the message emitted when a new activity is accepted.
type ActivityCreated struct {
	ActivityID   string    `json:"activity_id"`
	TenantID     string    `json:"tenant_id"`
	UserID       string    `json:"user_id"`
	ActivityType string    `json:"activity_type"`
	StartedAt    time.Time `json:"started_at"`
	DurationMin  int       `json:"duration_min"`
	Source       string    `json:"source"`
	Version      string    `json:"version"`
}

// ActivityStateChanged tracks state transitions (pending, synced, failed) for optimistic UI flows.
type ActivityStateChanged struct {
	ActivityID string    `json:"activity_id"`
	TenantID   string    `json:"tenant_id"`
	UserID     string    `json:"user_id"`
	State      string    `json:"state"`
	OccurredAt time.Time `json:"occurred_at"`
	Reason     string    `json:"reason,omitempty"`
}

File: libs/python/schemas/activity.py
"""Shared Pydantic models for activity domain events."""

from __future__ import annotations

from datetime import datetime
from enum import Enum
from pydantic import BaseModel, Field


class ActivityState(str, Enum):
    pending = "pending"
    synced = "synced"
    failed = "failed"


class ActivityCreated(BaseModel):
    activity_id: str = Field(..., alias="activity_id")
    tenant_id: str
    user_id: str
    activity_type: str
    started_at: datetime
    duration_min: int
    source: str
    version: str = "v1"


class ActivityStateChanged(BaseModel):
    activity_id: str
    tenant_id: str
    user_id: str
    state: ActivityState
    occurred_at: datetime
    reason: str | None = None

    class Config:
        populate_by_name = True
        use_enum_values = True
